ntoskrnl/ob/oblife.c

```diff
@@ -1558,6 +1558,9 @@ NtQueryObject(IN HANDLE ObjectHandle,
     POBJECT_HEADER ObjectHeader = NULL;
     POBJECT_HANDLE_ATTRIBUTE_INFORMATION HandleFlags;
     POBJECT_BASIC_INFORMATION BasicInfo;
+    POBJECT_TYPES_INFORMATION AllTypesInfo;
+    POBJECT_TYPE_INFORMATION ObjectTypeInfo;
+    ULONG ObjectTypeIndex;
```

Comment:
By curly-bracketing the whole "case" you are implementing, you could move these declarations inside the case-block.

Reply:
I would rather want to keep the current code structure style consistent with the rest of the `NtQueryObject` implementation, than diverging from the said current style. It would make sense to curly-bracketing all the information cases and have specific info case variables on each case body in a separate PR.